project1/application.py

```diff
@@ -116,3 +120,47 @@ def userHome(Username):
     if Username in session:
         return render_template("userDetails.html", username=Username, message="Successfully logged in.", heading="Welcome back")
     return redirect(url_for('index'))
+
+
+
+@app.route("/search",methods=["POST","GET"])
+def search():
+	if request.method == "GET":
+		return render_template("search.html")
+	else:
+		result = request.form.get("search")
+		result = '%'+result+'%'
+		search_result = Books.query.filter(or_(Books.tittle.ilike(result), Books.author.ilike(result), Books.isbn.ilike(result),Books.year.ilike(result))).all()
+		return render_template("search.html", books=search_result)
+
+
+@app.route("/bookpage/<username>/<isbn>", methods = ["POST","GET"])
+def bookpage(username, isbn):
+	user1 = username
```

Comment:
Check if the session varialble has a value or not, do not send the Username through the URI.

Reply:
I have already checked for username in session, but without username in URI it is not working we will implement that in integration part